lib/use_case/ai_response_board.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class AIResponse
+  def initialize(winning_combinations)
+    @winning_combinations = winning_combinations
+  end
+
+  def execute(board)
+    if board[4].nil?
+      4
+    elsif board[4] == 'X'
+      0
+    else
+      block_opponent(board)
+    end
+  end
+
+  private
+
+  def block_opponent(board)
+    @chance_to_block = []
+    @winning_combinations.each do |combination|
+      next if already_blocked(combination, board)
+
+      @chance_to_block << combination.select { |i| i if board[i] != 'X' }
```

Comment:
```suggestion
      @chance_to_block << combination.reject { |i| board[i] == 'X' }
```

Reply:
Good point! I think I should learn more about loops in Ruby. I never heard about .step before and .reject reads much better in this case.